### DISCUSSION TITLE
Slack integration is returnig invalid_auth

### DISCUSSION THREAD

**Discussion by chessbr** [requarks/wiki#3335] [Error / Bug Report]
**Describe the bug**
I configured the Slack integration according to the steps documented

**To Reproduce**
1. Just follow the steps for Slack integration from https://docs.requarks.io/auth/slack
2. Try to login using Slack
4. See the error

**Expected behavior**
The login should go through

**Screenshots**
![image](https://user-images.githubusercontent.com/8770370/109213484-6bf24400-778f-11eb-994a-27ebaf73bbe5.png)

**Host Info (please complete the following information):**
 - OS: Ubuntu 18.04
 - Wiki.js version: 2.5.170
 - Database engine: Postgres 12

**Additional context**
I've tried nearly everything, remove the auto registration, removed the domains, it won't work, althrough the users are listed as connected in Slack.

**Comment by chocobn69** [comment]
Hi, I faced exactly the same problem. Fresh install -> Slack integration -> invalid_auth
Using docker-compose Wiki.js (2.5.170)

**Comment by Corfus** [comment]
Hi, I can confirm the same problem, too.

In the debug-mode I got this feedback:
![Screenshot from 2021-03-05 23-43-17](https://user-images.githubusercontent.com/12698546/110182341-e153b980-7e0c-11eb-980d-fc999b378e34.png)

Perhaps something changed in the interface of Slack. I will try to find the origin of the issue.

**Comment by Corfus** [comment]
Okay I found the issue.
Slack needs a Post-Request to handle the Request: [https://api.slack.com/methods/users.identity](https://api.slack.com/methods/users.identity)

In the @aoberoi/passport-slack package it will be used a Get-Request, when you adapt the get to a post. It works fine.

[https://github.com/aoberoi/passport-slack/blob/5ccb3b40f294f88d170ce631554582665795dad2/src/strategy.js#L155](https://github.com/aoberoi/passport-slack/blob/5ccb3b40f294f88d170ce631554582665795dad2/src/strategy.js#L155)

Tomorrow I will start a Pull-Request to fix this in @aoberoi/passport-slack.

**Comment by ladygin-me** [comment]
Any news for fixing this issue?

**Reply by Corfus** [reply]
I started my PR (https://github.com/aoberoi/passport-slack/pull/11), but I fear we have to fork the project to get the fix

**Reply by NGPixel** [reply]
Unfortunately most of the passport projects are unmaintained or dead. I'm looking into cloning most of them for 3.0 and include them directly into the project instead.

**Reply by omenos** [reply]
I ran into this issue the other day, there's another more recently maintained Slack passport strategy I found here: https://github.com/nmaves/passport-slack-oauth2/

I'm not a JS person and haven't tried to vet it, but it may be worth a look.

**Reply by remdawg-mvsm** [reply]
@omenos Did you find a new to fix the issue or use the more recent Slack passport strategy  ?

**Reply by omenos** [reply]
@remdawg-mvsm I'm not a JS guy, so I wouldn't be of much help here. What I did was some research on what options were available and this seemed like a passport library that is:

- Recently supported and maintained
- Works with Slack's current auth messaging protocols

Two things that the currently used Slack passport integration lacks. I can't verify this information, but I think it may be something worth testing, particularly if the strategies are going to be vendored within the project in the next major version. And it would be preferable for existing strategies to be functional before that as well, if possible.

**Reply by ehossack** [reply]
@omenos  / @Corfus seems like the aforementioned nmaves implementation does work. I have raised a PR: https://github.com/Requarks/wiki/pull/4369
